src/OrbitBase/File.cpp

```diff
@@ -94,4 +112,25 @@ ErrorMessageOr<size_t> ReadFully(const unique_fd& fd, void* buffer, size_t size)
   return size - bytes_left;
 }
 
+ErrorMessageOr<size_t> ReadFullyAtOffset(const unique_fd& fd, void* buffer, size_t size,
+                                         off_t offset) {
+  uint8_t* current_position = static_cast<uint8_t*>(buffer);
+  size_t bytes_received = 0;
+  while (bytes_received < size) {
+    int64_t pread_result = TEMP_FAILURE_RETRY(pread(fd.get(), current_position, size, offset));
+    if (pread_result == -1) {
+      return ErrorMessage{SafeStrerror(errno)};
+    }
+    if (pread_result == 0) {
+      return bytes_received;
```

Comment:
break;

Reply:
Well, it doesn't matter, does it? Changed anyways.